juniper_official/Paragon/Northstar/notify_northstar.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+import requests
+import json
+import datetime
+import urllib3
+
+#NorthStar URLs
+node_get_url = 'http://ns-web.northstar:3301/NorthStar/API/v2/tenant/1/topology/1/nodes?q=$.[?(@.hostName %26%26 (@.hostName == "{hostname}"))]'
+link_get_url = 'http://ns-web.northstar:3301/NorthStar/API/v2/tenant/1/topology/1/links/?q=$.[?((@.endA%26%26(@.endA.node.id=="{node_id}")%26%26(@.endA.interfaceName=="{if_name}"))||(@.endZ%26%26(@.endZ.node.id=="{node_id}")%26%26(@.endZ.interfaceName=="{if_name}")))]'
+link_mnt_url = 'http://ns-web.northstar:3301/NorthStar/API/v2/tenant/1/topology/1/maintenances?q=$.[?(@.elements%26%26@.elements.filter(x => x.topoObjectType == "link" %26%26 x.index=="{link_index}").length >0 )]'
+mnt_url = 'http://ns-web.northstar:3301/NorthStar/API/v2/tenant/1/topology/1/maintenances'
+node_mnt_url = 'http://ns-web.northstar:3301/NorthStar/API/v2/tenant/1/topology/1/maintenances?q=$.[?(@.elements%26%26@.elements.filter(x => x.topoObjectType == "node" %26%26 x.index=={node_index}).length >0 )]'
+headers = {'Content-Type': 'application/json'}
+
+#Queries the NorthStar Node ID using device hostname.     
+def get_node_id(hostname, **kwargs):
+    r = requests.get(node_get_url.format(hostname=hostname), headers=headers, verify=False)
+    res = r.json()
+    if (r.status_code == 200) and (len(res)>0):
+        (node_id) = res[0]['id']
+        (node_index) = res[0]['nodeIndex']
+        return node_id, node_index
+    else:
+        raise Exception("Node doesn't exist")
+
+
+#Queries the NorthStar link ID using device hostname and IFD name and IFL number. 
+def get_link_index(hostname, ifd_name, ifl_number, **kwargs):
+    node_params = get_node_id(hostname)
+    if node_params:
+        node_id = node_params[0]
+        r = requests.get(link_get_url.format(node_id=node_id, if_name=ifd_name +"."+ ifl_number), headers=headers, verify=False)
```

Comment:
Please catch the exception raised by request.get.

Reply:
Added try except block